### PR TITLE
Add option to create channel using external SASL authentication method.

### PR DESCRIPTION
**Pull request by slepasteur**

### CODE REVIEW

File: src/SimpleAmqpClient/Channel.h [hunk 1]
```diff
@@ -91,6 +91,30 @@ class SIMPLEAMQPCLIENT_EXPORT Channel : boost::noncopyable {
...
+    * @param channel_max Request that the server limit the number of channels
+    * for
+    * this connection to the specified parameter, a value of zero will use the
+    * broker-supplied value
+    * @param frame_max Request that the server limit the maximum size of any
+    * frame to this value
+    * @return a new Channel object pointer
+    */
+  static ptr_t CreateSaslExternal(const std::string &host = "127.0.0.1", int port = 5672,
+                      const std::string &username = "guest",
```
Comment (alanxz) [inline src/SimpleAmqpClient/Channel.h:112]: To be consistent with rabbitmq-c API, this should be called 'identity'.

File: src/SimpleAmqpClient/Channel.h [hunk 2]
```diff
@@ -221,12 +287,12 @@ class SIMPLEAMQPCLIENT_EXPORT Channel : boost::noncopyable {
 
   explicit Channel(const std::string &host, int port,
                    const std::string &username, const std::string &password,
-                   const std::string &vhost, int frame_max);
+                   const std::string &vhost, int frame_max, bool sasl_external = false);
```
Comment (alanxz) [inline src/SimpleAmqpClient/Channel.h:290]: Don't use a default parameter. Update all callers instead, here and throughout.

File: src/SimpleAmqpClient/Channel.h (hunk 1 above)
Comment (slepasteur) [reply src/SimpleAmqpClient/Channel.h]: Done

File: src/SimpleAmqpClient/Channel.h (hunk 2 above)
Comment (slepasteur) [reply src/SimpleAmqpClient/Channel.h]: Done

**Pull request merged**